soc/st/stm32/stm32f7x/power.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2025 Ac6
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/pm.h>
+#include <clock_control/clock_stm32_ll_common.h>
+#include <soc.h>
+
+#include <stm32_ll_bus.h>
+#include <stm32_ll_cortex.h>
+#include <stm32_ll_pwr.h>
+
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+BUILD_ASSERT(DT_SAME_NODE(DT_CHOSEN(zephyr_cortex_m_idle_timer), DT_NODELABEL(rtc)),
+		"STM32Fx series needs RTC as an additional IDLE timer for power management");
+
+void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	switch (state) {
+	case PM_STATE_SUSPEND_TO_IDLE:
+		/* Prepare STOP mode with low-power regulator */
+		LL_LPM_DisableEventOnPend();
+		LL_PWR_ClearFlag_WU1();
+		LL_PWR_ClearFlag_WU2();
+		LL_PWR_ClearFlag_WU3();
+		LL_PWR_ClearFlag_WU4();
+		LL_PWR_ClearFlag_WU5();
+		LL_PWR_ClearFlag_WU6();
+
+		/* Use low-power regulator in STOP to reduce consumption */
+		LL_PWR_SetPowerMode(LL_PWR_MODE_STOP_LPREGU);
+		LL_LPM_EnableDeepSleep();
+
+		k_cpu_idle();
```

Comment:
calling `k_cpu_idle()` here is illegal (it unlocks IRQs)
```suggestion
		__WFI();
```